pyproject.toml

```diff
@@ -91,6 +91,14 @@ tox = "^4.4.7"
 twine = "^4.0.0"
 vulture = "^2.3"
 
+GPy = { version = "^1.10.0", optional = true }
+xgboost = { version = "^1.7.2", optional = true }
```

Comment:
I did not remember if this PR was written before poetry had support for [optional groups](https://python-poetry.org/docs/managing-dependencies/#optional-groups).

I then checked, and the "extras" syntax used here appears - indeed - to be the right one (from https://python-poetry.org/docs/master/managing-dependencies/):

> Dependency groups other than the implicit main group, must only contain dependencies you need in your development process. Installing them is only possible by using Poetry.
>
> To declare a set of dependencies, which add additional functionality to the project during runtime, use [extras](https://python-poetry.org/docs/master/pyproject/#extras) instead. Extras can be installed by the end user using pip.